qutip/qip/device/processor.py

```diff
@@ -542,13 +541,36 @@ def get_qobjevo(self, args=None, noisy=False):
             qu_list.append(qu)
 
         final_qu = _merge_qobjevo(qu_list)
-        final_qu.args.update(args)
+
+        final_qu.arguments(args)
 
         if noisy:
             return final_qu, c_ops
         else:
             return final_qu, []
 
+    def tlist(self, noisy):
+        """
+        Return the merged tlist of all the pulses.
+        
+        Parameters
+        ----------
+        noisy: bool, optional
+            If noise should be included. Default is False.
+        """
+        if not noisy:
+            dynamics = self.pulses
+        else:
+            dynamics = self.get_noisy_pulses(
+                device_noise=True, drift=True)
+
+        t_lists = []
+        for pulse in dynamics:
+            tlist = pulse.get_full_tlist()
+            if tlist is not None:
```

Comment:
Can `get_full_tlist` ever return anything other than a list? If `None` can be returned, perhapds `Drift.get_full_tlist` should return `None` instead?